modules/dnn/src/onnx/onnx_graph_simplifier.cpp

```diff
@@ -145,13 +156,266 @@ class SoftMaxSubgraph : public Subgraph
+
+        int scaleHW = addNodeToMatch("Scales", input);
+        int scaleNC = addNodeToMatch("Constant");
+        int cast1 = addNodeToMatch("Cast", scaleHW);
+
+        int shape = addNodeToMatch("Shape", input);
+        int slice = addNodeToMatch("Slice", shape);
+        int cast2 = addNodeToMatch("Cast", slice);
+
+        int inpDiv = addNodeToMatch("Div", cast1, cast2);
+        int inpConcat2 = addNodeToMatch("Concat", scaleNC, inpDiv);
+
+        addNodeToMatch("Upsample", input, inpConcat2);
+        setFusedNode("Upsample", input);
+    }
+
+    virtual bool match(const Ptr<ImportGraphWrapper>& net, int nodeId,
+                       std::vector<int>& matchedNodesIds,
+                       std::vector<int>& targetNodesIds) CV_OVERRIDE
+    {
+        if (FuseUpsampleSubgraph::match(net, nodeId, matchedNodesIds, targetNodesIds))
+        {
+            Ptr<ImportNodeWrapper> scales = net->getNode(matchedNodesIds[0]);
+            opencv_onnx::NodeProto* node = scales.dynamicCast<ONNXNodeWrapper>()->node;
+            opencv_onnx::AttributeProto attr = node->attribute(1);
+            CV_Assert(attr.name() == "scales");
+            CV_Assert(attr.ints_size() == 4);
+            scaleH = attr.ints(2);
+            scaleW = attr.ints(3);
+            return true;
+        }
+        return false;
+    }
+};
+
+class ResizeSubgraph : public FuseUpsampleSubgraph
+{
+public:
+    ResizeSubgraph()
+    {
+        int input = addNodeToMatch("");
+
+        int scaleHW = addNodeToMatch("Scales", input);
+        int constantROI = addNodeToMatch("Constant");
+
+        int shape = addNodeToMatch("Shape", input);
+        int axes_slice = addNodeToMatch("Constant");
+        int start_slice = addNodeToMatch("Constant");
+        int end_slice = addNodeToMatch("Constant");
+        int slice = addNodeToMatch("Slice", shape, start_slice, end_slice, axes_slice);
+
+        int cast = addNodeToMatch("Cast", scaleHW);
+
+        int concat = addNodeToMatch("Concat", slice, cast);
+        addNodeToMatch("Resize", input, constantROI, constantROI, concat);
+        setFusedNode("Upsample", input);
+    }
+
+    virtual bool match(const Ptr<ImportGraphWrapper>& net, int nodeId,
+                       std::vector<int>& matchedNodesIds,
+                       std::vector<int>& targetNodesIds) CV_OVERRIDE
+    {
+        if (FuseUpsampleSubgraph::match(net, nodeId, matchedNodesIds, targetNodesIds))
+        {
+            Ptr<ImportNodeWrapper> scales = net->getNode(matchedNodesIds[0]);
+            opencv_onnx::NodeProto* node = scales.dynamicCast<ONNXNodeWrapper>()->node;
+            opencv_onnx::AttributeProto attr = node->attribute(1);
+            CV_Assert(attr.name() == "scales");
+            CV_Assert(attr.ints_size() == 4);
+            scaleW = attr.ints(2);
+            scaleH = attr.ints(3);
+            //constant node id after matchedNodesIds removal.
+            emptyNodeId = matchedNodesIds.front();
```

Comment:
Why we need this? The node will be removed anyway or not?

Reply:
We need to remove this node. because this is empty constant node. `replace` function will not remove this.